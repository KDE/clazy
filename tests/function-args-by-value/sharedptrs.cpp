#include <QtCore/QSharedPointer>

struct Trivial {};

void test(QSharedPointer<int>) {}
void test(const QSharedPointer<int> &) {}
void test(Trivial ) {}
