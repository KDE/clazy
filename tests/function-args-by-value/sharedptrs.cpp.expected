function-args-by-value/sharedptrs.cpp:7:11: warning: test: Pass small and trivially-copyable type by value (const Trivial &) [-Wclazy-function-args-by-value]
